internal/workflow/replicator.go

```diff
@@ -148,6 +157,8 @@ func (w *Replicator) execute(ctx workflow.Context, request *ReplicatorRequest) e
 			reprocessChannel := workflow.NewNamedBufferedChannel(ctx, "replicator.reprocess", miniBatchCount)
 			defer reprocessChannel.Close()
 
+			responsesChannel := workflow.NewNamedBufferedChannel(ctx, "replicator.mini-batches.response", parallelism+miniBatchCount)
+
```

Comment:
need to make sure we close the channel by doing a defer responseChannel.Close() here.

Reply:
Sure. I will use this defer channel close instead of manually closing the channel